pinot-core/src/main/java/org/apache/pinot/core/common/BlockDocIdSet.java

```diff
@@ -29,4 +41,41 @@ public interface BlockDocIdSet {
    * ascending order.
    */
   BlockDocIdIterator iterator();
+
+  /**
+   * Returns the number of entries (SV value contains one entry, MV value contains multiple entries) scanned in the
+   * filtering phase. This method should be called after the filtering is done.
+   */
+  long getNumEntriesScannedInFilter();
+
+  /**
+   * For scan-based FilterBlockDocIdSet, pre-scans the documents and returns a non-scan-based FilterBlockDocIdSet.
+   */
+  default BlockDocIdSet toNonScanDocIdSet() {
```

Comment:
`toPreScannedDocIdSet`

Reply:
I kind of prefer the current name (carried from the existing code) since we have scan-based and non-scan-based iterator, and we want to convert the scan-based one to non-scan-based